packages/tasit-sdk/src/Decentraland.test.js

```diff
@@ -249,393 +314,657 @@ describe("Decentraland", () => {
+                );
+                done();
+              };
+
+              const errorListener = error => {
+                action.off("error");
+                done(error);
+              };
+
+              action.once("confirmation", confirmationListener);
+              action.on("error", errorListener);
+
+              action.send();
+            })();
+          }
+        );
 
-          await executeOrderAction.send();
-          await executeOrderAction.waitForOneConfirmation();
+        beforeEach(
+          "onboarding - approving Marketplace to spend ephemeral account's MANA",
+          done => {
+            (async () => {
+              // Global hooks don't run between same level hooks
+              await mineBlocks(provider, 1);
+
+              // Ephemeral account approves Marketplace to spend their mana tokens (actually tokens from Gnosis Safe).
+              mana.setWallet(ephemeralWallet);
+              const action = mana.approve(
+                MARKETPLACE_ADDRESS,
+                manaAmountForShopping
+              );
+
+              const confirmationListener = async message => {
+                const marketplaceAllowance = await mana.allowance(
+                  ephemeralAddress,
+                  MARKETPLACE_ADDRESS
+                );
+                expect(`${marketplaceAllowance}`).to.equal(
+                  `${manaAmountForShopping}`
+                );
+                done();
+              };
+
+              const errorListener = error => {
+                action.off("error");
+                done(error);
+              };
+
+              action.once("confirmation", confirmationListener);
+              action.on("error", errorListener);
+
+              action.send();
+            })();
+          }
+        );
 
-          await expectExactTokenBalances(estate, [ephemeralAddress], [1]);
+        it("should buy an estate", done => {
+          (async () => {
+            const {
+              assetId,
+              nftAddress,
+              seller,
+              priceInWei,
+              expiresAt,
+            } = estateForSale;
+
+            await checkAsset(estate, mana, estateForSale, GNOSIS_SAFE_ADDRESS);
+            await expectExactTokenBalances(estate, [ephemeralAddress], [0]);
+
+            const fingerprint = await estate.getFingerprint(`${assetId}`);
+
+            marketplace.setWallet(ephemeralWallet);
+            const action = marketplace.safeExecuteOrder(
+              nftAddress,
+              `${assetId}`,
+              `${priceInWei}`,
+              `${fingerprint}`
+            );
+
+            const confirmationListener = async message => {
+              // Allowance-of-allowance doesn't work
+              // See more: https://github.com/tasitlabs/TasitSDK/issues/273
+              //await expectExactTokenBalances(estate, [ephemeralAddress], [1]);
+              await expectExactTokenBalances(estate, [ephemeralAddress], [0]);
+              done();
```

Comment:
Is there any reason to call `done()` here if we know allowance-of-allowance doesn't work and we won't get to this path? Should we make it `done` with a new error here for now so the test won't pass if it gets here?

Reply:
I've understood that we want to force them pass (refs: https://github.com/tasitlabs/TasitSDK/issues/273#issuecomment-473977995).
Maybe be more explicitly on the comments? Related to https://github.com/tasitlabs/TasitSDK/pull/369#discussion_r280225416

Reply:
Yeah, we do want to force them to pass. But if we know ahead of time that the code will only flow through one path (the error path) and that's what we want to assert/expect is happening, couldn't we just have one done in the listener for the error path and not here (in the conf listener)?

Reply:
Ohh, got it. Yes, I agree. I'll remove the `done` from confirmationListener.

Reply:
Awesome, thanks!

Reply:
Done on https://github.com/tasitlabs/TasitSDK/pull/383

Reply:
Great, thanks! Resolving this